aspnetcore/blazor/call-web-api.md

```diff
@@ -15,20 +15,31 @@ By [Luke Latham](https://github.com/guardrex), [Daniel Roth](https://github.com/
 
 [!INCLUDE[](~/includes/blazorwasm-preview-notice.md)]
 
-Blazor WebAssembly apps call web APIs using a preconfigured `HttpClient` service. Compose requests, which can include JavaScript [Fetch API](https://developer.mozilla.org/docs/Web/API/Fetch_API) options, using Blazor JSON helpers or with <xref:System.Net.Http.HttpRequestMessage>.
+[Blazor WebAssembly](xref:blazor/hosting-models#blazor-webassembly) apps call web APIs using a preconfigured `HttpClient` service. Compose requests, which can include JavaScript [Fetch API](https://developer.mozilla.org/docs/Web/API/Fetch_API) options, using Blazor JSON helpers or with <xref:System.Net.Http.HttpRequestMessage>.
 
-Blazor Server apps call web APIs using <xref:System.Net.Http.HttpClient> instances typically created using <xref:System.Net.Http.IHttpClientFactory>. For more information, see <xref:fundamentals/http-requests>.
+[Blazor Server](xref:blazor/hosting-models#blazor-server) apps call web APIs using <xref:System.Net.Http.HttpClient> instances typically created using <xref:System.Net.Http.IHttpClientFactory>. For more information, see <xref:fundamentals/http-requests>.
 
-[View or download sample code](https://github.com/aspnet/AspNetCore.Docs/tree/master/aspnetcore/blazor/common/samples/) ([how to download](xref:index#how-to-download-a-sample))
+> [!NOTE]
+> This topic applies to Blazor WebAssembly apps. The JSON helpers described in this topic and shown in examples (for example, `GetJsonAsync`) aren't available for Blazor Server apps at this time. 
+>
+> For information on how to call web APIs from Blazor Server apps, see <xref:fundamentals/http-requests>, which explains how to use <xref:System.Text.Json?displayProperty=fullName> to serialize and deserialize JSON messages.
 
-For Blazor WebAssembly examples, see the following components in the sample app:
+[View or download sample code](https://github.com/aspnet/AspNetCore.Docs/tree/master/aspnetcore/blazor/common/samples/) ([how to download](xref:index#how-to-download-a-sample)) &ndash; Select the *BlazorWebAssemblySample* app.
+
+See the following components in the *BlazorWebAssemblySample* sample app:
 
 * Call Web API (*Pages/CallWebAPI.razor*)
 * HTTP Request Tester (*Components/HTTPRequestTester.razor*)
 
+## Package
```

Comment:
The default Blazor WebAssembly template references the Microsoft.AspNetCore.Blazor.HttpClient by default. This package is a .NET Standard library so it *can* be used from a Blazor Server app, but you have to add the package yourself. I believe the issues we've seen reported have come from customers trying to use this API from a Blazor Server app without having adding the package, correct?

Reply:
I thought everyone (including me ... I'll double-check in the morning) weren't getting the JSON helpers **_at all_** in a Blazor Server app even when this package **_is added_**. I'll check it again tomorrow morning.

Reply:
I seem to recall testing it when the topic went up, but I didn't re-test when I saw this ...

https://github.com/aspnet/AspNetCore.Docs/issues/15430#issuecomment-553644665

> GetAsJsonAsync is an API specific to the experimental package. You can use HttpContent.ReadAsAsync<T> as a substitute: https://stackoverflow.com/a/24052147.

... implies that you can't get the helpers at all in a Blazor Server app.

Reply:
Microsoft.AspNetCore.Blazor.HttpClient is an experimental package based on HttpClient + System.Text.Json. It should work anywhere .NET Standard 2.0 is supported, but it is experimental. If you want a stable API, then you can use Microsoft.AspNet.WebApi.Client, which is based on Json.NET.

Reply:
@danroth27 Couldn't wait until morning. :smile: I had to try it ... wouldn't :zzz: well.

I dropped the `CallWebAPI` component from the Blazor WebAssembly sample app into the Blazor Server sample app and provided it with the package reference ...

```xml
<Project Sdk="Microsoft.NET.Sdk.Web">

  <PropertyGroup>
    <TargetFramework>netcoreapp3.0</TargetFramework>
  </PropertyGroup>
  
  <ItemGroup>
    <PackageReference Include="Microsoft.AspNetCore.Blazor.HttpClient" Version="3.0.0-preview9.19465.2" />
  </ItemGroup>

</Project>
```

The component can be seen here :point_right: https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/blazor/common/samples/3.x/BlazorWebAssemblySample/Pages/CallWebAPI.razor

Cleaned and rebuilt the solution ... attempting to load the component, it throws the following ...

> An unhandled exception occurred while processing the request.
InvalidOperationException: Cannot provide a value for property 'Http' on type 'BlazorSample.Pages.CallWebAPI'. There is no registered service of type 'System.Net.Http.HttpClient'.
Microsoft.AspNetCore.Components.ComponentFactory+<>c__DisplayClass5_0.<CreateInitializer>g__Initialize|2(IServiceProvider serviceProvider, IComponent component)

Reply:
That error happens because Blazor Server apps don't have an `HttpClient` configured as a service by default like Blazor WebAssembly apps do. In a Blazor Server app, you have to create the `HttpClient` yourself or use the HttpClient factory infrastructure. This isn't related at all to the the JSON extension methods provided by the Microsoft.AspNetCore.Blazor.HttpClient package.